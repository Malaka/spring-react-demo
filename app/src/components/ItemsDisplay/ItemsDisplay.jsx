import React from 'react';
import {Button, ButtonGroup, Container, Table} from "reactstrap";
import {Link} from "react-router-dom";

export const ItemsDisplay = (props) => {

    const {items, removeHandler} = props;

    const itemList = items.map(item => {
        return (
            <tr key={item.id}>
                <td style={{whiteSpace: 'nowrap'}}>{item.name}</td>
                <td>{item.description}</td>
                <td>{item.category && <Button size="sm" tag={Link}
                                              to={`/search/${item.category.name}`}>{item.category.name}</Button>}</td>
                {removeHandler && <td>
                    <ButtonGroup>
                        <Button size="sm" color="primary" tag={Link} to={`/items/${item.id}`}>Edit</Button>
                        <Button size="sm" color="danger" onClick={() => removeHandler(item.id)}>Delete</Button>
                    </ButtonGroup>
                </td>}
            </tr>
        );
    });

    return (
        <div>
            <Container fluid>
                {props.children}
                <Table className="mt-4">
                    <thead>
                    <tr>
                        <th width="20%">Name</th>
                        <th>Description</th>
                        <th>Category</th>
                        {removeHandler && <th width="10%">Actions</th>}
                    </tr>
                    </thead>
                    <tbody>
                    {itemList}
                    </tbody>
                </Table>
            </Container>
        </div>
    );
};
