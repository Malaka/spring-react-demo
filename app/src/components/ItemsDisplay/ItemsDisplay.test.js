import * as React from "react";
import {shallow} from "enzyme";
import {ItemsDisplay} from "./ItemsDisplay";

describe("ItemsDisplay", () => {

    const items = [
        {
            id: 1,
            name: 'name1',
            description: 'description1',
            category: {
                name: 'food'
            }
        },
        {
            id: 2,
            name: 'name2',
            description: 'description2',
            category: {
                name: 'food'
            }
        }
    ];
    it("should render when passed handler", () => {
        const component = shallow(<ItemsDisplay items={items} removeHandler={() => {
        }}/>);
        expect(component).toMatchSnapshot();
    });

    it("should render when not pass handler", () => {
        const component = shallow(<ItemsDisplay items={items}/>);
        expect(component).toMatchSnapshot();
    });
});