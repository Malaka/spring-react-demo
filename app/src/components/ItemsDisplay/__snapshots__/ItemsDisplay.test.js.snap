// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemsDisplay should render when not pass handler 1`] = `
<div>
  <Container
    fluid={true}
    tag="div"
  >
    <Table
      className="mt-4"
      responsiveTag="div"
      tag="table"
    >
      <thead>
        <tr>
          <th
            width="20%"
          >
            Name
          </th>
          <th>
            Description
          </th>
          <th>
            Category
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          key="1"
        >
          <td
            style={
              Object {
                "whiteSpace": "nowrap",
              }
            }
          >
            name1
          </td>
          <td>
            description1
          </td>
          <td>
            <Button
              color="secondary"
              size="sm"
              tag={[Function]}
              to="/search/food"
            >
              food
            </Button>
          </td>
        </tr>
        <tr
          key="2"
        >
          <td
            style={
              Object {
                "whiteSpace": "nowrap",
              }
            }
          >
            name2
          </td>
          <td>
            description2
          </td>
          <td>
            <Button
              color="secondary"
              size="sm"
              tag={[Function]}
              to="/search/food"
            >
              food
            </Button>
          </td>
        </tr>
      </tbody>
    </Table>
  </Container>
</div>
`;

exports[`ItemsDisplay should render when passed handler 1`] = `
<div>
  <Container
    fluid={true}
    tag="div"
  >
    <Table
      className="mt-4"
      responsiveTag="div"
      tag="table"
    >
      <thead>
        <tr>
          <th
            width="20%"
          >
            Name
          </th>
          <th>
            Description
          </th>
          <th>
            Category
          </th>
          <th
            width="10%"
          >
            Actions
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          key="1"
        >
          <td
            style={
              Object {
                "whiteSpace": "nowrap",
              }
            }
          >
            name1
          </td>
          <td>
            description1
          </td>
          <td>
            <Button
              color="secondary"
              size="sm"
              tag={[Function]}
              to="/search/food"
            >
              food
            </Button>
          </td>
          <td>
            <ButtonGroup
              role="group"
              tag="div"
            >
              <Button
                color="primary"
                size="sm"
                tag={[Function]}
                to="/items/1"
              >
                Edit
              </Button>
              <Button
                color="danger"
                onClick={[Function]}
                size="sm"
                tag="button"
              >
                Delete
              </Button>
            </ButtonGroup>
          </td>
        </tr>
        <tr
          key="2"
        >
          <td
            style={
              Object {
                "whiteSpace": "nowrap",
              }
            }
          >
            name2
          </td>
          <td>
            description2
          </td>
          <td>
            <Button
              color="secondary"
              size="sm"
              tag={[Function]}
              to="/search/food"
            >
              food
            </Button>
          </td>
          <td>
            <ButtonGroup
              role="group"
              tag="div"
            >
              <Button
                color="primary"
                size="sm"
                tag={[Function]}
                to="/items/2"
              >
                Edit
              </Button>
              <Button
                color="danger"
                onClick={[Function]}
                size="sm"
                tag="button"
              >
                Delete
              </Button>
            </ButtonGroup>
          </td>
        </tr>
      </tbody>
    </Table>
  </Container>
</div>
`;
